projects/netgrif-components-core/src/lib/panel/task-panel-list/task-panel-list-pagination/abstract-task-list-pagination.component.ts

```diff
@@ -0,0 +1,43 @@
+import {Inject, Input, Optional} from '@angular/core';
+import {Observable} from 'rxjs';
+import {tap} from 'rxjs/operators';
+import {ActivatedRoute} from '@angular/router';
+import {AbstractDefaultTaskList} from '../default-task-panel-list/abstract-default-task-list';
+import {TaskPanelData} from '../task-panel-data/task-panel-data';
+import {TaskViewService} from '../../../view/task-view/service/task-view.service';
+import {LoggerService} from '../../../logger/services/logger.service';
+import {NAE_TAB_DATA} from '../../../tabs/tab-data-injection-token/tab-data-injection-token';
+import {InjectedTabData} from '../../../tabs/interfaces';
+
+export abstract class AbstractTaskListPaginationComponent extends AbstractDefaultTaskList {
+
+    public length: number;
+    public pageSize = 20;
+    public pageIndex = 0;
+    public pageSizeOptions: Array<number> = [10, 20, 50];
+
+    @Input()
+    set tasks$(tasks: Observable<Array<TaskPanelData>>) {
+        this._tasks$ = tasks.pipe((tap(() => {
```

Comment:
who subscribes to this Observable? We had problems in the past with calls inside a `pipe()` not being executed if the subscription was done in HTML with the `async` angular pipe (`|`) operator

Reply:
this solution works, and it was done like this because it wasn't working when pipe was only  in constructor, this is workaround for problem you write about